autoload/go/lint.vim

```diff
@@ -18,17 +14,23 @@ if !exists("g:go_errcheck_bin")
     let g:go_errcheck_bin = "errcheck"
 endif
 
-function! go#lint#Gometa(path_to_lint) abort
+function! go#lint#Gometa(...) abort
+    if a:0 == 0
+        let goargs = expand('%:p:h')
```

Comment:
You only pass the current file's path. Previously we were recursively looking for all files under the current buffer's directory (with `./...`). This means our `:GoMetaLinter` functionality has changed. I've added `g:go_metalinter_path` just for that, because for my case I want that it goes and lints all files. 

I think I don't get an error because of this simple line in my vimrc:

```
autocmd BufEnter * silent! lcd %:p:h
```

So the fix here is to return absolute paths from gometalinter, not passing a folder just to make it return absolute paths.